lib/rgeo/shapefile/reader.rb

```diff
@@ -342,7 +343,10 @@ def next
       # and yield the Reader::Record for each one.
 
       def each
+        return to_enum(:each) unless block_given?
         return unless @opened
```

Comment:
Here I guess we should raise then, otherwise this would lead to a very weird behaviour.